### PR TITLE
Poetry 1.2

### PR DESCRIPTION
**Pull request by vwxyzjn**
## Description
Adopting poetry 1.2 support. The locking and adding dependencies are orders of magnitude faster!

## Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] New algorithm
- [ ] Documentation

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/vwxyzjn/cleanrl/blob/master/CONTRIBUTING.md) guide (**required**).
- [x] I have ensured `pre-commit run --all-files` passes (**required**).
- [x] I have updated the documentation and previewed the changes via `mkdocs serve`.
- [x] I have updated the tests accordingly (if applicable).

### CODE REVIEW

**Comment by vwxyzjn** [comment]
@Dipamc77 and @dosssman this PR is ready for review.

**Comment by dosssman** [comment]
@vwxyzjn Hello there.
Attempted to test the workflow from scratch by installing poetry 1.2.1 as per their docs using this command: `curl -sSL https://install.python-poetry.org | python3 -`, but running `poetry install` in cleanrl directory while under the `poetry1.2` branch just hangs.
Is it on my side only ?
```
$ poetry install
Creating virtualenv cleanrl-iXm2iEAu-py3.9 in /home/d055/.cache/pypoetry/virtualenvs
Installing dependencies from lock file

Package operations: 67 installs, 1 update, 0 removals

  • Installing certifi (2022.6.15.1): Pending...
  • Installing charset-normalizer (2.1.1): Pending...
  • Installing idna (3.3): Pending...
  • Installing pyasn1 (0.4.8): Pending...
  • Installing pyparsing (3.0.9): Pending...
  • Installing urllib3 (1.26.12): Pending...
```

Or is my luck just that bad today ? :sweat_smile:

**Comment by vwxyzjn** [comment]
@dosssman could you try `export PYTHON_KEYRING_BACKEND=keyring.backends.null.Keyring`? I think it's related to https://github.com/python-poetry/poetry/issues/1917

**Comment by dosssman** [comment]
Thanks a lot. This seems to have done the trick now.

File: docs/get-started/installation.md [hunk 1]
```diff
@@ -29,30 +37,38 @@ poetry install
...
-poetry install -E procgen
-poetry install -E envpool
-poetry install -E pettingzoo
+poetry install --with atari
+poetry install --with pybullet
+poetry install --with mujoco
+poetry install --with procgen
+poetry install --with envpool
+poetry install --with pettingzoo
 ```
```
Comment (dosssman) [inline docs/get-started/installation.md:72]: Also add 
```
poetry install -E jax
poetry install -E optuna
```
for the relevant scenarios.

**Review by dosssman — APPROVED** [review]
Tested, looking good over here.
Great work.

**Review by kinalmehta — APPROVED** [review]
Hello,
Test done as follows:
- Installation successful
- Did not check if every script ran perfectly. Only tested one for each environment (atari, gym, mujoco, envpool, procgen) and framework (JAX and PyTorch)

LGTM.

**Pull request merged**